tour_model_eval/Explore multiple datasets.ipynb

```diff
@@ -0,0 +1,947 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a48cede4",
+   "metadata": {},
+   "source": [
+    "### Entries with low cluster trip ratio and high request pct"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "acb3f8f2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fig = plt.Figure(figsize=(20,5))\n",
+    "axarr = fig.subplots(1,2)\n",
+    "colors = plt.get_cmap(\"Accent\", 4).colors\n",
+    "\n",
+    "for i, algo in enumerate([\"dbscan\", \"oursim\"]):\n",
+    "    ax = axarr[i]\n",
+    "    for j, p in enumerate(participant_df.program.unique()):\n",
+    "        curr_p_df = participant_df[participant_df.program==p]\n",
+    "        # Focus on trips with > 10 entries for meaningful results\n",
+    "        curr_p_df[curr_p_df.n_labeled_trips > 10].plot.scatter(x=f\"request_pct_{algo}\", y=f\"cluster_trip_ratio_{algo}\", ax=ax, color=colors[j], label=p)\n",
+    "fig"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8352a785",
+   "metadata": {},
+   "source": [
+    "While the oursim is essentially a straight line, there are more outliers for DBSCAN.\n",
+    "let's look at some of the outliers for DBSCAN."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "43adb310",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "participant_df.query(\"request_pct_dbscan > 0.5 and cluster_trip_ratio_dbscan < 0.3 and n_labeled_trips > 10\")[[\"program\", \"n_labeled_trips\", \"n_trip_in_cluster_dbscan\", \"n_clusters_dbscan\", \"request_pct_dbscan\", \"cluster_trip_ratio_dbscan\"]]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "59bc1c5b",
+   "metadata": {},
+   "source": [
+    "Looking at the entries with the biggest differences:\n",
+    "- **71f21d53**:\n",
+    "  - clusters (2) + trips outside cluster (34) = 36. So request pct is 36/48 = 0.75\n",
+    "  - clusters (2) / trips in cluster (14) = 0.14\n",
+    "  \n",
+    "So basically, the original assumption was correct; it is the noisy points that make a difference. Basically, this happens in the case where there are many noisy clusters, but the points that are within the clusters, compress down well. In that case, the noisy trips carry a lot more weight in the request pct, and excluding them has a much higher impact.\n",
+    "\n",
+    "- **a3587c47**:\n",
+    "  - clusters (2) + trips outside cluster (5) = 7. So request pct should be 7/12 = 0.58. **Why is this 0.66 instead?!**\n",
+    "  - clusters (2) / trips in cluster (7) = 0.28\n",
+    "  \n",
+    "There was a mistake in the n_clusters calculation. I was using the max of the column, but the labels start from zero, so the number of clusters is actually the max + 1.\n",
+    "\n",
+    "Fixed in notebook. New calculation is:\n",
+    "- **a3587c47**:\n",
+    "  - clusters (3) + trips outside cluster (5) = 8. So request pct should be 8/12 = 0.66.\n",
+    "  - clusters (3) / trips in cluster (7) = 0.42\n",
```

Comment:
I don't see `a3587c47` in the table.

Reply:
How many entries do you see in the table? I can't verify this until I finish evaluating the mapping changes.

Reply:
5, one from minipilot, four from stage.

Reply:
Please see the note below it.

> There was a mistake in the n_clusters calculation. I was using the max of the column, but the labels start from zero, 
> so the number of clusters is actually the max + 1.
> Fixed in notebook. New calculation is:
...

After the new calculation, for `a3587c47`, clusters (3) / trips in cluster (7) = 0.42 so it doesn't match the query and does not show up. If you change the query you should be able to see it, and see that the `cluster_trip_ratio_dbscan` is 0.42.

Feel free to update this in the notebook and submit a PR to this branch.